File: code/12-JavaCodeLoop/JavaTDDLoop/src/test/java/com/example/testsupport/FakeSecretNumberGenerator.java
package com.example.testsupport;

import com.example.valueobjects.Interval;
import com.example.domain.SecretNumberGenerator;

/**
 * Deterministic test double for {@code SecretNumberGenerator}.
 *
 * <p>Always returns the fixed secret number it was created with, regardless of the
 * interval passed in. Used so that {@code GuessGame} behaviour can be tested without
 * relying on real randomness.</p>
 */
public class FakeSecretNumberGenerator extends SecretNumberGenerator {

    private final int fixedSecret;

    public FakeSecretNumberGenerator(int fixedSecret) {
        this.fixedSecret = fixedSecret;
    }

    @Override
    public int generate(Interval interval) {
        return fixedSecret;
    }
}

File: code/13-Java2Python/python/tests/testsupport/fake_secret_number_generator.py
from domain.secret_number_generator import SecretNumberGenerator
from valueobjects.interval import Interval


class FakeSecretNumberGenerator(SecretNumberGenerator):
    """Deterministic test double for SecretNumberGenerator.

    Always returns the fixed secret number it was created with, regardless
    of the interval passed in. Used so that GuessGame behaviour can be
    tested without relying on real randomness.
    """

    def __init__(self, fixed_secret: int) -> None:
        super().__init__()
        self._fixed_secret = fixed_secret

    def generate(self, interval: Interval) -> int:
        return self._fixed_secret
